bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/WorkflowDelegationStep.java

```diff
@@ -0,0 +1,161 @@
+import com.adobe.granite.workflow.model.WorkflowModel;
+import com.day.cq.commons.inherit.HierarchyNodeInheritanceValueMap;
+import com.day.cq.commons.inherit.InheritanceValueMap;
+
+
+/**
+ * 
+ * This workflow steps just calls another workflow as subworkflow.
+ * 
+ * The subworkflow is determined by looking up the correct workflow on the 
+ * resource (or its ascendents). Works on pages and assets.
+ * 
+ * In order to support multiple instances of this behaviour, the name of the property,
+ * which determines the model of the subworkflow, can be be configured as arguments on
+ * the workflow step.
+ * 
+ * Arguments should be provided as this:
+ * 
+ * workflowModelProperty=<propName>,defaultWorkflowModel=<pathToDefaultWorkflowModel>
+ * 
+ * where <propName> is the name of the property which contains the paths of the workflow
+ * models
+ * 
+ * and <pathToDefaultWorkflowModel> the path to the default Workflow Model, which is used
+ * as fallback.
+ * 
+ * If no workflow model can be resolved (directly and as fallback), a workflow exception is thrown.
+ * 
+ *
+ */
+
+@Service()
+@Component()
+@Properties({
+    @Property(
+            label = "Workflow Label",
+            name = "process.label",
+            value = "Workflow Delegation step",
+            description = "Calls configurable sub workflows"
+    )
+})
+public class WorkflowDelegationStep implements WorkflowProcess {
+	
+	
+	@Reference
+	ResourceResolverFactory rrf;
+
+	private static final Logger LOG = LoggerFactory.getLogger(WorkflowDelegationStep.class);
+	
+	// Under this property the workflow model is stored
+	private static final String WORKFLOW_PROPERTY_NAME = "workflowModelProperty";
+	
+	// A default workflow model can be provided as fallback
+	private static final String DEFAULT_WORKFLOW_MODEL = "defaultWorkflowModel";
+	
+	@Override
+	public void execute(WorkItem item, WorkflowSession session, MetaDataMap metadata)
+			throws WorkflowException {
+		
+		Map<String,String> args = getProcessArgsMap(metadata);
+		
+		String propertyName = args.get(WORKFLOW_PROPERTY_NAME);
+		if (StringUtils.isBlank(propertyName)) {
+			throw new WorkflowException ("propertyname not defined");
+		}
+		String defaultWorkflowModel = args.get(DEFAULT_WORKFLOW_MODEL);
+		
+		LOG.debug ("propertyName = {}, default workflow model = {}", 
+				new Object[]{propertyName,defaultWorkflowModel});
+		
+		ResourceResolver resolver = null;
+		WorkflowData wfData = item.getWorkflowData();
+		if (wfData.getPayloadType().equals("JCR_PATH")) {
+			String path = wfData.getPayload().toString();
+			
+			Session jcrSession = session.adaptTo(Session.class);
+			Map<String, Object> map = new HashMap<String, Object>();
+			map.put("user.jcr.session", jcrSession);
+			try {
+				resolver = rrf.getResourceResolver(map);
+			} catch (LoginException e) {
+				throw new WorkflowException (e);
+			}
+			
+			if (resolver != null) {
+				Resource wfResource = resolver.getResource(path);
```

Comment:
I only need a resource here to use the InheritanceValueMap, which works flawlessly (so it does also consider the properties of the jcr:content subnodes of ancestors). I just tested it with pages and assets and it works fine.